File: angular/src/app/utility/region.service.ts
import { Injectable } from "@angular/core";

@Injectable()
export class RegionService {

    private regions: RegionType[] = [
        RegionType.WESTERN_EUROPE,
        RegionType.EASTERN_EUROPE,
        RegionType.CHINA,
        RegionType.SOUTHEAST_ASIA,
        RegionType.NORTH_AMERICA,
        RegionType.SOUTH_AMERICA,
    ];

    public getAllRegions(): RegionType[] {
        return this.regions;
    }

    public getRegionByIndex(index: number): RegionType {
        return this.regions[index];
    }
}

export class RegionType {
    static readonly WESTERN_EUROPE  = new RegionType("WESTERN_EUROPE", 0, "Western Europe", "WEU");
    static readonly EASTERN_EUROPE = new RegionType("EASTERN_EUROPE", 1, "Eastern Europe", "EEU");
    static readonly CHINA  = new RegionType("CHINA", 2, "China", "CN");
    static readonly SOUTHEAST_ASIA  = new RegionType("SOUTHEAST_ASIA", 3, "SE Asia", "SEA");
    static readonly NORTH_AMERICA = new RegionType("NORTH_AMERICA", 4, "North America", "NA");
    static readonly SOUTH_AMERICA  = new RegionType("SOUTH_AMERICA", 5, "South America", "SA");

    private constructor(private readonly key: string,
        public readonly index: number,
        public readonly value: string,
        public readonly abbreviation: string) {
    }

    public toString(): string {
        return this.key;
    }
}

File: src/springboot/src/main/java/wiwitaditya/demo/dotapedia/db/utility/RegionService.java
package wiwitaditya.demo.dotapedia.db.utility;

import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@Component
public class RegionService {

    public List<String> getAllRegions() {
        List<String> result = new ArrayList();
        result.add(Region.WESTERN_EUROPE.toString());
        result.add(Region.EASTERN_EUROPE.toString());
        result.add(Region.CHINA.toString());
        result.add(Region.SOUTHEAST_ASIA.toString());
        result.add(Region.NORTH_AMERICA.toString());
        result.add(Region.SOUTH_AMERICA.toString());
        return result;
    }

    public List<String> toRegions(Region region) {
        List<String> result = new ArrayList();
        result.add(region.toString());
        return result;
    }
}
